packages/ui/components/ui/icon-input.tsx

```diff
@@ -1,21 +1,18 @@
 import { ReactNode } from 'react';
-import { Input } from './input';
+import { Input, InputProps } from './input';
 
-/**
- * Use this to render an input with an icon to its left, such as a search field
- * with a magnifying glass.
- */
-export const IconInput = ({
-  value,
-  onChange,
-  icon,
-  placeholder,
-}: {
+interface IconInputProps extends Omit<InputProps, 'onChange'> {
```

Comment:
issue: we shouldn't be allowing all props to pass through to underlying components in our component design system

This is part of a bigger discussion*, but in short, we've had issues with our component system becoming quite messy because any props can be passed through. For example, this change to the props interface now allows `children`, `className`, etc. to be passed to `<IconInput />`. `children` doesn't make sense to pass to it, and `className` allows consumers to drastically change the appearance of `<IconInput />` in ways they shouldn't be able to.

Instead, please just add `autoFocus` as a prop; we can keep adding more props as we need them, rather than allowing all props by default.

\* I'm soon going to be converting #432 to a tracking issue so that we can gradually shift our design system components away from our current model.

Reply:
Thanks for the detailed explanation, it really helped me understand the idea! Changed this to only add `autoFocus`